src/app/tracing/configuration/filter-table-view/filter-table-view.component.ts

```diff
@@ -256,27 +256,49 @@ export class FilterTableViewComponent implements OnChanges, DoCheck, OnInit, OnD
         this.selectColumns.emit();
     }
 
-    onRowSelectionChange({ selected }: { selected: TableRow[] }): void {
-        if (this.processedInput__) {
-            const selectedRowIds = selected.map(row => row.id);
-            // dblclick events trigger 3 selection change events
-            // only the first one changes (usually) the selection
-            // we check here the selection change to emit only true selection changes
-            if (this.areSelectedRowIdsEqual(selectedRowIds, this.processedInput__.selectedRowIds)) {
-                // no selection change
-            } else {
-                // selection change detected
-                this.processedInput__.selectedRowIds = selected.map(row => row.id);
+    isTableRowArray(toInspect: TableRow[] | { selected: TableRow[] } | any): toInspect is TableRow[] {
+        return (toInspect as TableRow[]).length !== undefined;
+    }
 
-                this.selectedRows_.splice(0, this.selectedRows_.length);
-                this.selectedRows_.push(...selected);
+    isTableRowObject(toInspect: TableRow[] | { selected: TableRow[] } | any): toInspect is { selected: TableRow[] } {
+        return (toInspect as { selected: TableRow[] }).selected !== undefined;
+    }
 
-                this.rowSelectionChange.emit(this.processedInput__.selectedRowIds);
+    onRowSelectionChange(eventIn: { selected: TableRow[] } | TableRow[] | any): void {
+        // Early Return in case we are not ready to work on this yet.
+        if (!this.processedInput__) {
+            return;
+        }
 
-                // we need this to get rid of the text selection
-                window.getSelection()?.removeAllRanges();
-            }
+        // Typeguards to ensure that rows is TableRow[] in following code
+        let rows: TableRow[] = [];
+        if (this.isTableRowObject(eventIn)) {
+            rows = eventIn.selected;
+        } else if (this.isTableRowArray(eventIn)) {
```

Comment:
Optional:
- the test `Array.isArray(eventIn)` would also be sufficient here for me